templates/static/js/example.js

```diff
@@ -79,136 +140,232 @@
+   * @param {HTMLElement} placementElement The placeholder element for the example
+   * @returns {Promise<{rendered: String, body: String}>} The rendered HTML and source code of the example
+   */
+  async function fetchHtmlSource(placementElement) {
+    const renderedHtml = await fetchResponseText(placementElement.href);
+    const bodyHTML = getExampleSection('body', renderedHtml);
 
-    var height = placementElement.getAttribute('data-height');
+    return {rendered: renderedHtml, body: bodyHTML};
+  }
 
-    var codeSnippet = document.createElement('div');
+  /**
+   * Fetches the raw Jinja template of an example and returns the Jinja content block
+   * @param {HTMLElement} placementElement The placeholder element for the example
+   * @returns {Promise<String>} The Jinja content block of the example
+   */
+  async function fetchJinjaContentBlock(placementElement) {
+    // Raw templates are not served at standalone paths, so strip it from the URL if it was found.
+    const exampleUrl = new URL(`${placementElement.href.replace(/standalone/, '/')}`);
 
+    // Add `?raw=true` query parameter to the URL to request the raw Jinja template
+    const queryParams = new URLSearchParams(exampleUrl.search);
+    queryParams.set('raw', true);
+    exampleUrl.search = queryParams.toString();
+
+    const rawJinjaTemplate = await fetchResponseText(exampleUrl.toString());
+    return formatSource(getExampleSection('jinja', rawJinjaTemplate), 'jinja');
+  }
+
+  /**
+   * Replaces an example placeholder element with its rendered result and code snippet.
+   * @param {HTMLAnchorElement} placementElement `a.js-example` element used as a placeholder for the example to render
+   * @param {String} renderedHtml Full document HTML of the example as it is shown to end-users
+   * @param {String|null} jinjaTemplate Jinja Template of the example as it may be used by developers, if supported
+   */
+  async function renderExample(placementElement) {
+    const codeSnippet = document.createElement('div');
     codeSnippet.classList.add('p-code-snippet', 'is-bordered');
 
-    var header = document.createElement('div');
+    const header = document.createElement('div');
     header.classList.add('p-code-snippet__header');
-    var titleEl = document.createElement('h5');
+
+    const titleEl = document.createElement('h5');
     titleEl.classList.add('p-code-snippet__title');
 
-    // example page title is structured as "... | Examples | Vanilla documentation"
-    // we want to strip anything after first | pipe
-    titleEl.innerText = title.split('|')[0];
+    const srcData = {
+      html: undefined,
+      jinja: undefined,
+      css: undefined,
+      js: undefined,
+      codePen: undefined,
+    };
 
-    header.appendChild(titleEl);
-    codeSnippet.appendChild(header);
+    const exampleRequests = [];
+
+    const fetchHtml = fetchHtmlSource(placementElement).then(({rendered: renderedHtml, body: bodyHtml}) => {
```

Comment:
Why do we parse `body` within `fetchHtmlSource`, while we have to parse `title` and `head` separately anyway?

I guess for consitency it would make sense to either, do all the parsing inside `fetchHTMLSource` (and return object with all the fields we are interested in), or make `fetchHTMLSource` return only the raw `renderedHTML`, so that all parts of it will be parsed explicitly after. Or is there a reason for `body` to be treated differently?

Reply:
It was originally structured the first way you suggested (return all the fields) but I abandoned it to avoid returning a huge object to the promise callback. However I prefer it to parsing the HTML in the callback, so I've changed it back to do all the extraction in `fetchHtmlSource` (renamed to `fetchRenderedHtml`)